src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -247,6 +241,981 @@ public bool NeedsRefresh()
             return false;
         }
 
+        public void ResetState()
+        {
+            _featureFlagIndex = 0;
+
+            return;
+        }
+
+        private FormatException CreateFeatureFlagFormatException(string jsonPropertyName, string settingKey, string foundJsonValueKind, string expectedJsonValueKind)
+        {
+            return new FormatException(string.Format(
+                ErrorMessages.FeatureFlagInvalidJsonProperty,
+                jsonPropertyName,
+                settingKey,
+                foundJsonValueKind,
+                expectedJsonValueKind));
+        }
+
+        private FeatureFlag ParseFeatureFlag(string settingKey, string settingValue)
```

Comment:
I don't like us having to do this manually. Did we look into source generation as an alternative? https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/reflection-vs-source-generation?pivots=dotnet-8-0#reflection

Reply:
That's fair, it does require a lot more testing and is more code to maintain.  One reason to continue with this approach is it follows the pattern we established in our other .NET repos of always using Utf8JsonReader for reading Json, which is more efficient compared to the JsonSerializer.Deserialize method. Also, we already added it to the main branch without variants/allocation/telemetry and this PR is re-adding it for merge conflict purposes. 

I'm not sure, but from what I can tell [here](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation?pivots=dotnet-8-0) it looks like source generation isn't available in all of the versions we support, since it's only for C# 9.0 and later, and .NET 6.0 and later.